lib/metadataTypes/MetadataType.js

```diff
@@ -287,29 +300,42 @@ class MetadataType {
                             'Hotfix for non-cachable resource found in deploy folder. Trying update:'
                         );
                         Util.logger.warn(JSON.stringify(metadata[metadataKey]));
-                        metadataToUpdate.push({
-                            before: {},
-                            after: metadata[metadataKey],
-                        });
+                        if (error) {
+                            metadataToUpdate.push(null);
+                        } else {
+                            metadataToUpdate.push({
+                                before: {},
+                                after: metadata[metadataKey],
+                            });
+                        }
                     } else if (cache.getByKey(this.definition.type, normalizedKey)) {
                         // normal way of processing update files
                         metadata[metadataKey][this.definition.idField] = cache.getByKey(
                             this.definition.type,
                             normalizedKey
                         )[this.definition.idField];
-                        metadataToUpdate.push({
-                            before: cache.getByKey(this.definition.type, normalizedKey),
-                            after: metadata[metadataKey],
-                        });
+                        if (error) {
+                            // do this in case something went wrong during pre-deploy steps to ensure the total counter is correct
+                            metadataToUpdate.push(null);
```

Comment:
so if we are keeping count, we should put another var above the for loop, not putting in null.

Reply:
why not?

Reply:
we would need to additional variables to count updates and creates in parallel. simply pushing null makes it a lot cleaner